gui/include/mainwindow.h

```diff
@@ -0,0 +1,195 @@
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+********************************************************************************/
+#ifndef MAINWINDOW_H
+#define MAINWINDOW_H
+
+#include <string>
+#include <iostream>
+#include <sstream>
+
+#include <boost/make_unique.hpp>
+
+#include <opencv2/core/types.hpp>
+
+#include <QMainWindow>
+#include <QDebug>
+#include <QFileDialog>
+#include <QTextStream>
+#include <QMessageBox>
+#include <QValidator>
+#include <QCloseEvent>
+#include <QStatusBar>
+#include <QProgressBar>
+
+#include <OpenSpaceNetArgs.h>
+#include <OpenSpaceNet.h>
+#include <processthread.h>
+#include <imagery/MapServiceClient.h>
+
+#include "progresswindow.h"
+#include "qdebugstream.h"
+
+namespace Ui {
+    class MainWindow;
+}
+
+class MainWindow : public QMainWindow
+{
+    Q_OBJECT
+
+public:
+    explicit MainWindow(QWidget *parent = 0);
+    ~MainWindow();
+
+private slots:
+    void on_loadConfigPushButton_clicked();
+
+    void on_saveConfigPushButton_clicked();
+
+    void on_runPushButton_clicked();
+
+    void on_localImageFileBrowseButton_clicked();
+
+    void on_modelFileBrowseButton_clicked();
+
+    void on_outputLocationBrowseButton_clicked();
+
+    void on_viewMetadataButton_clicked();
+
+    void on_helpPushButton_clicked();
+
+    void on_nmsCheckBox_toggled(bool checked);
+
+    void on_bboxOverrideCheckBox_toggled(bool checked);
+
+    void enableRunButton();
+
+    void updateProgressBox(QString updateText);
+
+    void on_imageSourceComboBox_currentIndexChanged(const QString &source);
+
+    void on_modelpathLineEditLostFocus();
+
+    void on_imagepathLineEditLostFocus();
+
+    void on_mapIdLineEditCursorPositionChanged();
+
+    void on_tokenLineEditCursorPositionChanged();
+
+    void on_passwordLineEditCursorPositionChanged();
+
+    void on_usernameLineEditCursorPositionChanged();
+
+    void on_outputLocationLineEditLostFocus();
+
+    void on_localImagePathLineEditCursorPositionChanged();
+
+    void on_modelpathLineEditCursorPositionChanged();
+
+    void on_outputFilenameLineEditCursorPositionChanged();
+
+    void on_outputPathLineEditCursorPositionChanged();
+
+    void on_closePushButton_clicked();
+
+    void cancelThread();
+
+private:
+    void connectSignalsAndSlots();
+    void setUpLogging();
+    void initValidation();
+
+    void resetProgressWindow();
+    void closeEvent(QCloseEvent *event);
+    bool eventFilter(QObject *obj, QEvent *event);
+
+    bool validateUI(QString &error);
+    void importConfig(QString configFile);
+    void exportConfig(const QString &filepath);
+
+    dg::osn::OpenSpaceNetArgs osnArgs;
```

Comment:
Are we double saving things?  It looks like there is an Args member, and then other members that hold the same data.